### DISCUSSION TITLE
How to redefined command by macors?

### DISCUSSION THREAD

**Discussion by lesonli** [KaTeX/KaTeX#3139] [Q&A]
```
macros: {
      '\\hat':'\\htmlStyle{font-family: xxx;}{\\hat { #1 }}'
}
```
ParseError: KaTeX parse error: Too many expansions: infinite loop or need to increase maxExpand setting

**Comment by edemaine** [comment]
The problem with your macro is that you're redefining `\hat`, but within the macro, you're calling `\hat`. Hence you get an infinite loop.

You could instead call your macro `\Hat` to avoid the issue:

```js
macros: {
      '\\Hat':'\\htmlStyle{font-family: xxx;}{\\hat { #1 }}'
}
```

**Reply by lesonli** [reply]
Thank you~

Because many latex formulas have been saved in the document, but we need to adjust the rendering effect of the formula. 
If we define another command, we must modify the code of all formulas. 

Therefore, I think whether there is a parameter in the macro definition, which can be redefined only once without triggering the loop definition.
